### PR TITLE
Flaky PoC

### PR DESCRIPTION
**Pull request by oleorhagen**
This is just a test for verifying the use of the flaky package in order to rerun
tests which fail.

The PoC is added only to 'test_demo_artifact' to begin with, as this is a common
culprit of pipeline failures.

Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

**Comment by oleorhagen** [comment]
Follow up from the QA meeting.

[Flaky GitHub](https://github.com/box/flaky)

Easy to enable, and should be handy for the regular culprits I think.

File: tests/tests/test_demo_artifact.py [hunk 1]
```diff
@@ -102,6 +103,7 @@ def run_demo_script_up(exit_cond=exit_cond):
 
     # Give the test a timeframe, as the script might run forever,
     # if something goes awry, or the script is not brought down properly.
+    @flaky(max_runs=3)
```
Comment (lluiscampos) [inline tests/tests/test_demo_artifact.py:106]: We should always have a JIRA ticket associated with any test marked as `flaky`. A comment should be enough.

**Review by lluiscampos — COMMENTED** [review]
Good research! I looked the project in GitHub and it looks okey.

I vote for extending this to all known unstable tests (only the ones that have JIRA tasks [here](https://tracker.mender.io/browse/QA-276)) and merge soon.

File: tests/tests/test_demo_artifact.py (hunk 1 above)
Comment (oleorhagen) [reply tests/tests/test_demo_artifact.py:106]: Great idea!

**Comment by oleorhagen** [comment]
First run ready for:
* test_demo_artifact (3-reruns)
* test_enterprise_mtls (3-reruns)
* test_enterprise_mtls_hsm (3-reruns)


Let's see how it goes

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline please

**Review by lluiscampos — APPROVED** [review]

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline please :pray:

**Comment by oleorhagen** [comment]
Output looks pretty nice:
```
===Flaky Test Report===

test_demo_artifact passed 1 out of the required 1 times. Success!
test_mtls_enterprise_hsm[rsa] passed 1 out of the required 1 times. Success!

===End Flaky Test Report===
================ 15 passed, 437 warnings in 1437.97s (0:23:57) =================
```

I did missplace one of the flaky tags though, so rerunning one more time...

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

**Pull request merged**